flink-end-to-end-tests/flink-stream-state-ttl-test/src/main/java/org/apache/flink/streaming/tests/verify/ValueWithTs.java

```diff
@@ -56,8 +57,8 @@ public String toString() {
 	/** Serializer for Serializer. */
 	public static class Serializer extends CompositeSerializer<ValueWithTs<?>> {
 
-		public Serializer(TypeSerializer<?> userValueSerializer) {
-			super(true, userValueSerializer, LongSerializer.INSTANCE);
+		public Serializer(TypeSerializer<?> valueSerializer, TypeSerializer<Long> timestampSerializer) {
```

Comment:
I don't think we need a public constructor that accepts the timestamp serializer.
This should be a private constructor used only by the snapshot class.

We should still have a public constructor that accepts the user value serializer, and by default just uses `LongSerializer.INSTANCE` as the new timestamp serializer.

Reply:
As far as I can tell this is not a user facing serializer, rather used internally by the `TtlSerializer`,
and I think it is important to make it explicit that this is a composite serializer and these are the nested serializers that define it.

As a way to reduce verbosity, we can add a static factory method with an explicit name.
Feel free to decide for yourself :-)

Reply:
ok, that's a fair argument that I can agree with. Since I don't have a strong opinion on this, I'll leave the constructors as is in this PR.